File: trip-service-java/src/main/java/com/cawadev/softwarecraft/katas/tripservice/trip/TripDAO.java
package com.cawadev.softwarecraft.katas.tripservice.trip;

import com.cawadev.softwarecraft.katas.tripservice.exception.CollaboratorCallException;
import com.cawadev.softwarecraft.katas.tripservice.user.User;

import java.util.List;

public class TripDAO {

    public static List<Trip> findTripsByUser(User user) {
        throw new CollaboratorCallException(
                "TripDAO should not be invoked on an unit test.");
    }
}

File: trip-service-kotlin/src/main/kotlin/com/cawadev/softwarecraft/katas/tripservice/trip/TripDAO.kt
package com.cawadev.softwarecraft.katas.tripservice.trip

import com.cawadev.softwarecraft.katas.tripservice.exception.CollaboratorCallException
import com.cawadev.softwarecraft.katas.tripservice.user.User

class TripDAO {

    companion object {
        @JvmStatic
        fun findTripsByUser(user: User): List<Trip> {
            throw CollaboratorCallException("TripDAO should not be invoked on an unit test.")
        }
    }
}
